internal/supervisor/import.go

```diff
@@ -44,90 +49,293 @@ func NewImportSupervisor(store importStore, logger log.FieldLogger, cloudClient
 		if err != nil {
-			logger.WithError(err).Error("Failed to mark import as completed")
+			logger.WithError(err).Error("Failed to update import")
 			return
 		}
+		return
+	}
 
-		logger.Info("Import completed")
+	logger = s.logger.WithFields(log.Fields{
+		"installation": installation.ID,
+	})
 
-		if s.keepImportData {
-			logger.Debug("Skipping import bundle cleanup")
-			return
-		}
+	newState := s.transitionImport(imp, installation, logger)
 
-		cfg, err := config.LoadDefaultConfig(context.TODO())
+	if newState != imp.State {
+		imp.State = newState
+		err := s.store.UpdateImport(imp)
 		if err != nil {
-			logger.WithError(err).Error("Failed to load AWS config")
+			logger.WithError(err).Error("Failed to update import")
 			return
 		}
+	}
+}
 
-		key := fmt.Sprintf("%s.zip", i.TranslationID)
-		ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second*5)
-		defer cancelFunc()
+func (s *ImportSupervisor) transitionImport(imp *model.Import, installation *cloud.InstallationDTO, logger log.FieldLogger) string {
+	switch imp.State {
+	case model.ImportStateRequested:
+		return s.transitionImportRequested(imp, installation, logger)
+	case model.ImportStateInstallationPreAdjustment:
+		return s.transitionImportInstallationPreAdjustment(imp, installation, logger)
+	case model.ImportStateInProgress:
+		return s.transitionImportInProgress(imp, installation, logger)
+	case model.ImportStateComplete:
+		return s.transitionImportComplete(imp, installation, logger)
+	case model.ImportStateInstallationPostAdjustment:
+		return s.transitionImportInstallationPostAdjustment(imp, installation, logger)
+	}
 
-		client := s3.NewFromConfig(cfg)
-		_, err = client.DeleteObject(ctx, &s3.DeleteObjectInput{
-			Bucket: &s.bucket,
-			Key:    &key,
-		})
-		if err != nil {
-			logger.WithError(err).Error("Failed to delete translation from S3")
-			return
+	return imp.State
+}
+
+func (s *ImportSupervisor) transitionImportRequested(imp *model.Import, installation *cloud.InstallationDTO, logger log.FieldLogger) string {
+
+	if installation.State != cloud.InstallationStateStable {
+		logger.Debugln("Waiting for installation to be stable")
+		return imp.State
+	}
+
+	var adjustmentRequired bool
+	size := installation.Size
+	if size != model.Size1000String {
+		logger.Infof("Resizing installation to %s\n", model.Size1000String)
+		size = model.Size1000String
+		adjustmentRequired = true
+	}
+
+	s3TimeoutEnv := installation.MattermostEnv[model.S3EnvKey]
+	if s3TimeoutEnv.Value != fmt.Sprintf("%d", model.S3ExtendedTimeout) {
+		logger.Infoln("Extending S3 timeout to 48 hours")
+		s3TimeoutEnv.Value = fmt.Sprintf("%d", model.S3ExtendedTimeout)
+		adjustmentRequired = true
+	}
+
+	if !adjustmentRequired {
+		logger.Infoln("No adjustments required")
+		return model.ImportStateInProgress
+	}
+
+	logger.Infoln("Adjustments required")
+	patchInstallation := &cloud.PatchInstallationRequest{
+		Size: &size,
+		MattermostEnv: cloud.EnvVarMap{
```

Comment:
We are going to want to use `PriorityEnv` for our env changes so that we ensure we take priority over any group-level env var settings.
```suggestion
		PriorityEnv: cloud.EnvVarMap{
```